latex/slides/07_arrays.tex

```diff
@@ -350,40 +348,58 @@ \subsection{}
 		\item \textit{stdin} is the standard input
 	\end{itemize}
 \end{frame}
-\begin{frame}[fragile]{Array of arrays}
+\begin{frame}[fragile]{Passing arrays}
+	An array is passed by its identifier, no big deal. The real question is, how a function is defined, that takes an array as argument.
+	\begin{lstlisting}[numbers=none]
+int foo(char arr[]) {
+	printf("%d\n", arr[0]);
+}
+\end{lstlisting}
+	\begin{itemize}
+		\item \textbf{But:} If you want to get the size of the array in this function, you'll get a false value. The reason for this is discussed later.
+		\item So you may want to pass the size as a second argument
+		\item This prevents out-of-bounds problems
```

Comment:
Maybe add one more bullet point like
* If the size of the array is unknown, you never know how many elements you can access safely

Which explains why out-of-bounds accesses can occur.